plenum/server/node.py

```diff
@@ -2509,17 +2616,22 @@ def initStateFromLedger(self, state: State, ledger: Ledger, reqHandler):
 
     def initDomainState(self):
         self.initStateFromLedger(self.states[DOMAIN_LEDGER_ID],
-                                 self.domainLedger, self.reqHandler)
+                                 self.domainLedger, self.get_req_handler(DOMAIN_LEDGER_ID))
 
     def addGenesisNyms(self):
         # THIS SHOULD NOT BE DONE FOR PRODUCTION
         for _, txn in self.domainLedger.getAllTxn():
             if txn.get(TXN_TYPE) == NYM:
                 self.addNewRole(txn)
 
-    def defaultAuthNr(self):
+    def init_core_authenticator(self):
         state = self.getState(DOMAIN_LEDGER_ID)
-        return SimpleAuthNr(state=state)
+        return CoreAuthNr(state=state)
+
+    def defaultAuthNr(self) -> ReqAuthenticator:
+        req_authnr = ReqAuthenticator()
+        req_authnr.register_authenticator(self.init_core_authenticator())
```

Comment:
What other authenticators are we going to have?
How is it related to plugins?

Reply:
Plugins can have their own code for authentication